internal/app/coroutines/claimTask.go

```diff
@@ -0,0 +1,215 @@
+package coroutines
+
+import (
+	"log/slog"
+
+	"github.com/resonatehq/resonate/internal/kernel/metadata"
+	"github.com/resonatehq/resonate/internal/kernel/scheduler"
+	"github.com/resonatehq/resonate/internal/kernel/t_aio"
+	"github.com/resonatehq/resonate/internal/kernel/t_api"
+	"github.com/resonatehq/resonate/internal/util"
+	"github.com/resonatehq/resonate/pkg/task"
+)
+
+func ClaimTask(metadata *metadata.Metadata, req *t_api.Request, res CallBackFn) *Coroutine {
+	return scheduler.NewCoroutine(metadata, func(c *Coroutine) {
+		// Read task.
+		rc, err := readTask(c, req, req.ClaimTask.TaskId)
+		if err != nil {
+			res(nil, err)
+			return
+		}
+
+		util.Assert(rc.Store != nil, "completion must not be nil")
+		readTaskResult := rc.Store.Results[0].ReadTask
+		util.Assert(readTaskResult.RowsReturned == 0 || readTaskResult.RowsReturned == 1, "result must return 0 or 1 rows")
```

Comment:
nit: if `readTaskResult.RowsReturned` can't be negative then `readTaskResult.RowsReturned <= 1` would be a less verbose option.

btw, `readTaskResult := rc.Store.Results[0].ReadTask`, is `rc.Store.Results` guarantee to always be non-empty?

Reply:
Thank you @ahmedtadde! 

Regarding the first nit, that totally makes sense. As for the second point, our completion object (variable `rc` in this case), is a "global" object that contains [all possible completions](https://github.com/resonatehq/resonate/blob/main/internal/kernel/t_aio/aio.go#L44-L49) across all of our AIO subsystems. As such, all keys may potentially be unset with the exception of Kind, which is supposed to indicate which entry is set but this is not guaranteed by the type system.

I think it would make sense to shore up these assertions across all of our [coroutines](https://github.com/resonatehq/resonate/tree/main/internal/app/coroutines)
1. check all possibly nil keys
2. ensure the key that is set corresponds to kind
3. ensure that no other key is set other than the one that corresponds to kind

This might be a good first time contributor PR, let me know if you're interested :)